pkg/osquery/tables/firefox_preferences/table.go

```diff
@@ -0,0 +1,85 @@
+package firefox_preferences
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go"
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	name   string
+	logger log.Logger
+}
+
+const tableName = "kolide_firefox_preferences"
+
+func TablePlugin(_ *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+	)
+
+	t := &Table{
+		name:   tableName,
+		logger: logger,
+	}
+
+	return table.NewPlugin(t.name, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	return generateData(queryContext, t.logger)
+}
+
+func generateData(queryContext table.QueryContext, logger log.Logger) ([]map[string]string, error) {
+	paths := tablehelpers.GetConstraints(queryContext, "path")
+
+	if len(paths) != 1 {
+		return nil, errors.Errorf("The %s table requires that you specify a constraint for path", tableName)
+	}
+
+	file, err := os.Open(paths[0])
+	if err != nil {
+		// TODO: Investigate what error message looks like. Add filepath possibly
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(file)
+
+	rowData := map[string]string{"path": paths[0]}
+	rawKeyVals := make(map[string]interface{})
+
+	re := regexp.MustCompile(`user_pref\((.*)\)`)
+	for scanner.Scan() {
+		line := scanner.Text()
+		match := re.FindStringSubmatch(line)
+
+		if len(match) <= 1 {
+			continue
+		}
+
+		parts := strings.Split(match[1], ", ")
```

Comment:
What happens if there are multiple commas? Suggest adding some to the test cases

Reply:
Are any of them user inputted? What happens for things like:
```
user_pref("app.update.background.previous.reasons", "[\"the maintenance service registry, has a comma\"]");
```

basically, you're looking at the impossibility of parsing complex nested data with regexps. There may be some clever things -- you know the larger line structure. But here be dragons

Reply:
Good point, didn't think about that. I don't think there is any place for a user to input data, but I'll have to look closer. If this is the case, then maybe regex isn't the way to go here, and we want to use the parsing library after all?